ring/ingester_lifecycle.go

```diff
@@ -117,8 +117,10 @@ func (r *IngesterRegistration) pickTokens() []uint32 {
 
 func (r *IngesterRegistration) heartbeat(tokens []uint32) {
 	heartbeat := func(in interface{}) (out interface{}, retry bool, err error) {
-		ringDesc := &Desc{}
-		if in != nil {
+		var ringDesc *Desc
+		if in == nil {
+			ringDesc = newDesc()
```

Comment:
c.f. https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare

No action required.